### PR TITLE
Change tree view from select to modal on listview

### PR DESCRIPTION
**Pull request by adudekyeti**
image
After click, show modal:

image

### CODE REVIEW

File: modules/Vtiger/views/RecordsList.php [hunk 1]
```diff
@@ -81,6 +81,7 @@ public function postProcessAjax(App\Request $request)
 	public function getModalScripts(App\Request $request)
 	{
 		return array_merge(parent::getModalScripts($request), $this->checkAndConvertJsScripts([
+			'~layouts/resources/Fields.js',
```
Comment (rskrzypczak) [inline modules/Vtiger/views/RecordsList.php]: ?

File: public_html/layouts/basic/modules/Vtiger/resources/ListSearch.js [hunk 2]
```diff
@@ -135,6 +135,12 @@ jQuery.Class(
 		registerDateTimeListSearch: function(container) {
 			App.Fields.DateTime.register(this.getContainer());
 		},
+		registerTreeSearch: function() {
+			App.Fields.Tree.register(this.container);
+			let vtigerEditInstance = new Vtiger_Edit_Js();
+			vtigerEditInstance.registerTreeAutoCompleteFields(this.container);
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/ListSearch.js]: move registerTreeAutoCompleteFields, registerClearTreeSelectionEvent to App.Fields.Tree

File: modules/Vtiger/uitypes/Tree.php [hunk 3]
```diff
@@ -51,15 +51,27 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
 			return '';
 		}
 		$fieldModel = $this->getFieldModel();
-		if ($rawText) {
-			$text = \App\Fields\Tree::getPicklistValue($fieldModel->getFieldParams(), $fieldModel->getModuleName())[$value];
-			if (is_int($length)) {
-				$text = \App\TextParser::textTruncate($text, $length);
+		if (false === strpos($value, ',')) {
```
Comment (rskrzypczak) [inline modules/Vtiger/uitypes/Tree.php:54]: ?

File: layouts/basic/modules/Vtiger/Modals/RecordsList.tpl [hunk 4]
```diff
@@ -38,6 +38,16 @@
 									&nbsp;&nbsp;<span class="fas {if $ORDER_BY[$LISTVIEW_HEADER_NAME] eq \App\Db::DESC}fa-chevron-down{else}fa-chevron-up{/if}"></span>
 								{/if}
 							</span>
+							{if $LISTVIEW_HEADER->getFieldDataType() eq 'tree' || $LISTVIEW_HEADER->getFieldDataType() eq 'categoryMultipicklist'}
+								{assign var=LISTVIEW_HEADER_NAME value=$LISTVIEW_HEADER->getName()}
+								<div class="d-flex align-items-center">
+									<input name="searchInSubcategories" value="1" type="checkbox" class="searchInSubcategories mr-1" id="searchInSubcategories{$LISTVIEW_HEADER_NAME}" title="{\App\Language::translate('LBL_SEARCH_IN_SUBCATEGORIES',$MODULE_NAME)}" data-columnname="{$LISTVIEW_HEADER->getColumnName()}" {if !empty($SEARCH_DETAILS[$LISTVIEW_HEADER_NAME]['specialOption'])} checked {/if}>
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/Modals/RecordsList.tpl]:
```suggestion
								<div class="d-flex align-items-center">
									<input name="searchInSubcategories" value="1" type="checkbox" class="searchInSubcategories mr-1" id="searchInSubcategories{\App\Layout::getUniqueId()}" title="{\App\Language::translate('LBL_SEARCH_IN_SUBCATEGORIES',$MODULE_NAME)}" data-columnname="{$LISTVIEW_HEADER->getColumnName()}" {if !empty($SEARCH_DETAILS[$LISTVIEW_HEADER->getName()]['specialOption'])} checked {/if}>
```

**Review by rskrzypczak — CHANGES_REQUESTED** [review]

File: modules/Vtiger/uitypes/Tree.php (hunk 3 above)
Comment (adudekyeti) [reply modules/Vtiger/uitypes/Tree.php:54]: ?

**Pull request merged**